ports/mimxrt/sdcard.c

```diff
@@ -0,0 +1,989 @@
+static bool sdcard_cmd_send_csd(mimxrt_sdcard_obj_t *card, csd_t *csd);
+static bool sdcard_cmd_select_card(mimxrt_sdcard_obj_t *sdcard);
+static bool sdcard_cmd_set_blocklen(mimxrt_sdcard_obj_t *sdcard);
+static bool sdcard_cmd_set_bus_width(mimxrt_sdcard_obj_t *sdcard, uint8_t bus_width);
+
+
+// ---
+void sdcard_card_inserted_callback(USDHC_Type *base, void *userData) {
+    #if defined MICROPY_USDHC1 && USDHC1_AVAIL
+    if (base == USDHC1) {
+        sdcard_usdhc1_inserted = true;
+
+    }
+    #endif
+    #if defined MICROPY_USDHC2 && USDHC2_AVAIL
+    if (base == USDHC2) {
+        sdcard_usdhc2_inserted = true;
+
+    }
+    #endif
+
+    USDHC_ClearInterruptStatusFlags(base, kUSDHC_CardInsertionFlag);
+}
+
+void sdcard_card_removed_callback(USDHC_Type *base, void *userData) {
+    #if defined MICROPY_USDHC1 && USDHC1_AVAIL
+    if (base == USDHC1) {
+        sdcard_usdhc1_inserted = false;
+
+    }
+    #endif
+    #if defined MICROPY_USDHC2 && USDHC2_AVAIL
+    if (base == USDHC2) {
+        sdcard_usdhc2_inserted = false;
+
+    }
+    #endif
+
+    USDHC_ClearInterruptStatusFlags(base, kUSDHC_CardRemovalFlag);
+}
+
+void sdcard_transfer_complete_callback(USDHC_Type *base, usdhc_handle_t *handle, status_t status, void *userData) {
+    sdcard_transfer_status = status;
+    sdcard_transfer_done = true;
+    USDHC_ClearInterruptStatusFlags(base, kUSDHC_CommandCompleteFlag | kUSDHC_DataCompleteFlag);
+}
+
+void sdcard_dummy_callback(USDHC_Type *base, void *userData) {
+    return;
+}
+
+static status_t sdcard_transfer_blocking(USDHC_Type *base, usdhc_handle_t *handle, usdhc_transfer_t *transfer, uint32_t timeout_ms) {
+    uint32_t retry_ctr = 0UL;
+    status_t status;
+
+    usdhc_adma_config_t dma_config;
+
+    (void)memset(&dma_config, 0, sizeof(usdhc_adma_config_t));
+    dma_config.dmaMode = kUSDHC_DmaModeAdma2;
+    dma_config.burstLen = kUSDHC_EnBurstLenForINCR;
+    dma_config.admaTable = sdcard_adma_descriptor_table;
+    dma_config.admaTableWords = DMA_DESCRIPTOR_BUFFER_SIZE;
+
+    do
+    {
+        status = USDHC_TransferNonBlocking(base, handle, &dma_config, transfer);
+        retry_ctr++;
+    } while (!(status == kStatus_Success) && (retry_ctr < 1000000UL));
+
+    if (retry_ctr > retry_ctr_max) {
+        retry_ctr_max = retry_ctr;
+    }
+
+    if (status == kStatus_Success) {
+        for (int i = 0; i < timeout_ms * 100; i++) {
+            if ((sdcard_transfer_done == true) && (sdcard_transfer_status == kStatus_Success)) {
+                sdcard_transfer_done = false;
+                return kStatus_Success;
+            }
+            ticks_delay_us64(10);
+        }
+        return kStatus_Timeout;
+    } else {
+        return status;
+    }
+}
+
+static void sdcard_decode_csd(mimxrt_sdcard_obj_t *card, csd_t *csd) {
+    uint8_t csd_structure = 0x3 & (csd->data[3] >> 30);
+
+    uint8_t read_bl_len;
+    uint32_t c_size;
+    uint8_t c_size_mult;
+
+    switch (csd_structure)
+    {
+        case (0): {
```

Comment:
no need for parenthesis around the integer, ie use `case 0: {`

Reply:
fixed.